pkg/cluster/cluster.go

```diff
@@ -409,6 +409,23 @@ func createClusterSpec(config Spec, awsClient aws.Client) (*cmv1.Cluster, error)
 		}
 	}
 
+	if config.AvailabilityZones != nil {
+		clusterBuilder = clusterBuilder.Nodes(
```

Comment:
We already set `clusterBuilder.Nodes` above (line 375), I think you should add this code inside the `clusterNodesBuilder` code:
```
if config.ComputeMachineType != "" || config.ComputeNodes != 0 || if config.AvailabilityZones != nil {
  clusterNodesBuilder := cmv1.NewClusterNodes()
  // etc...
}